src/rust/ide/view/graph-editor/src/component/node.rs

```diff
@@ -689,21 +686,25 @@ impl Node {
             eval bg_color ((c) model.background.bg_color.set(c.into()));
 
 
-            // === Tooltip ===
+            // === Output Label ===
 
-            // Hide tooltip if we show the preview vis.
-            frp.source.tooltip <+ preview_visible.on_true().constant(tooltip::Style::unset_label());
-            // Propagate output tooltip. Only if it is not hidden, or to disable it.
-            block_tooltip      <- hide_tooltip && has_tooltip;
-            frp.source.tooltip <+ model.output.frp.tooltip.gate_not(&block_tooltip);
+            model.output.set_port_label_visibility
+                <+ visualization_visible.not().and(&no_error_set);
 
 
             // === VCS Handling ===
+
             model.vcs_indicator.frp.set_status <+ frp.set_vcs_status;
         }
-
+        
         model.error_visualization.set_layer(visualization::Layer::Front);
-        frp.set_error.emit(None);
+
+        // Those three lines initialize the values that are necessary for port labels to show.
+        // All three are needed in this order, for reasons that I do not understand.
+        frp.set_error(None);
+        frp.enable_visualization();
+        frp.disable_visualization();
```

Comment:
This part seems problematic. But I don't know how we initialize networks properly.

Reply:
Well, the root cause is that each frp node has output being an output value until some input events arrives.

Knowing that, let's look at the line: 
```
            model.output.set_port_label_visibility
                <+ visualization_visible.not().and(&no_error_set);
```
the visualization_visible is properly false, however no_error_set could not - until someone emits signal about new error. I'm not sure why disabling/enabling visualizations is necessary - @wdanilo do you have any thoughts?

If we want to have port types displayed by default, we should directly fire model.output.set_port_label_visibility with true during initialization

Reply:
Firing `model.output.set_port_label_visibility` might not change anything since it would get overwritten by the network connection that you mentioned. I'll have to find the roots from where the incorrect values propagate.

Generally, I find it a bit problematic that I have to care about the consistent initialization of networks. The connections that I create already describe how it should be initialized. Is there a better way to do this?

Reply:
Well, as I know, the FRP node output value is not overwritten be setting up network connection _per se_, but only when the event go through it. So we need to intialize the output value to have proper value before any input connection will be fired.